scripts/data_processing/PPI/ngs_subnational_capacity_data.R

```diff
@@ -0,0 +1,86 @@
+#### Create NGS subnational capacity data file ####
+#### Jordan Klein
+
+#### 1) Setup ####
+#### Run gisaid metadata extraction & processing script
+source("gisaid_metadata_geokeys_processing.R")
+## Update file linking gisaid locations to shapefiles & population w/ new gisaid metadata
+source("geokey_id_update.R")
+
+#### Clear environment & load metadata
+rm(list = ls())
+Metadata_raw <- read_csv("../../../data/raw/metadata.csv") # from extracted datastream
+
+##### Get set of unique geolocations present in the metadata
+#### Split location string into geographies at each level (up to 7)
+Locations_split <-  str_split(Metadata_raw$location, "  |/", simplify = T) %>% 
+    as_tibble() %>%
+    mutate(across(everything(), ~str_trim(.))) %>%
+    mutate(across(everything(), ~ifelse(.=="", NA, as.character(.))))
+colnames(Locations_split) <- paste0("geo_l", 1:7)
+
+### Combine back with metadata
+Metadata <- bind_cols(Metadata_raw, Locations_split)
+# give each location a unique id
+Metadata <-tibble(location = unique(Metadata$location), 
+                  loc_id = seq(1, nrow(unique(select(Metadata, location))))) %>% 
+    full_join(Metadata)
+
+### Clear memory & unneeded objects
+rm(Metadata_raw)
+rm(Locations_split)
+gc()
+
+#### Load packages
+library("devtools")
+library("subregionalcovid19")
+
+#### Read in joined Pop-Geo file & IDed unique geokeys in metadata
+Geo.Pop <- st_read("../../../data/raw/GeoPop.shp")
+Geokeys_IDed <- read_csv('../../../data/raw/geo_keys_ids.csv', col_types = "dcccccccc")
+
```

Comment:
I am sure you have given this some thought already, but how do you think about handling these one off fixes at scale? Is there a way to create like a data dictionary or is it pretty complex because they are all at different levels?

Note, I did the same thing in my processing file for the one off countries, just curious if you have ideas on a more scalable solution, as I imagine this manual editing will become impossible as we go down in granularity

Reply:
This could be dealt with by the basemap or database having defined geometries for all the different levels of subdivisions. So we could just map the geocoded gisaid locations to these geometries. A lot of these one-off fixes have to do with GA tech's geometries being weird for some countries.